examples/user_interfaces/embedding_in_tk_sgskip.py

```diff
@@ -7,57 +7,44 @@
 import matplotlib
 matplotlib.use('TkAgg')
 
-from numpy import arange, sin, pi
-from matplotlib.backends.backend_tkagg import FigureCanvasTkAgg, NavigationToolbar2TkAgg
-# implement the default mpl key bindings
+import numpy as np
+from matplotlib.backends.backend_tkagg import (
+    FigureCanvasTkAgg, NavigationToolbar2TkAgg)
+# Implement the default Matplotlib key bindings.
 from matplotlib.backend_bases import key_press_handler
-
-
 from matplotlib.figure import Figure
+from six.moves import tkinter as Tk
 
-import sys
-if sys.version_info[0] < 3:
-    import Tkinter as Tk
-else:
-    import tkinter as Tk
 
 root = Tk.Tk()
-root.wm_title("Embedding in TK")
-
-
-f = Figure(figsize=(5, 4), dpi=100)
-a = f.add_subplot(111)
-t = arange(0.0, 3.0, 0.01)
-s = sin(2*pi*t)
+root.wm_title("Embedding in Tk")
 
-a.plot(t, s)
+fig = Figure(figsize=(5, 4), dpi=100)
+t = np.arange(0, 3, .01)
+fig.add_subplot(111).plot(t, 2 * np.sin(2 * np.pi * t))
 
-
-# a tk.DrawingArea
-canvas = FigureCanvasTkAgg(f, master=root)
-canvas.show()
+canvas = FigureCanvasTkAgg(fig, master=root)  # A tk.DrawingArea.
+canvas.draw()
 canvas.get_tk_widget().pack(side=Tk.TOP, fill=Tk.BOTH, expand=1)
 
 toolbar = NavigationToolbar2TkAgg(canvas, root)
 toolbar.update()
 canvas._tkcanvas.pack(side=Tk.TOP, fill=Tk.BOTH, expand=1)
 
-
-def on_key_event(event):
-    print('you pressed %s' % event.key)
-    key_press_handler(event, canvas, toolbar)
-
-canvas.mpl_connect('key_press_event', on_key_event)
+canvas.mpl_connect(
```

Comment:
some debugging code?

Reply:
Either way works but I guess the previous one looks a bit better.  Reverting.